### PR TITLE
fix(curriculum): Remove additional unnecessary assert message argument from English Coding Interview Prep challenges

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x]  My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Related to issue #36400

This PR removes assert message arguments from `testString` for the `Coding Interview Prep` curriculum section.  This PR was created to remove assert message arguments which were wrapped in two single quotes instead of single quotes.  [See this discussion for more information](https://github.com/freeCodeCamp/freeCodeCamp/pull/36409#issuecomment-510219004)

### CODE REVIEW

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-join.english.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ tests:
...
-    testString: 'assert.deepEqual(hashJoin(hash1, hash2), res, ''<code>hashJoin([{ age: 27, name: "Jonah" }, { age: 18, name: "Alan" }, { age: 28, name: "Glory" }, { age: 18, name: "Popeye" }, { age: 28, name: "Alan" }], [{ character: "Jonah", nemesis: "Whales" }, { character: "Jonah", nemesis: "Spiders" }, { character: "Alan", nemesis: "Ghosts" }, { character:"Alan", nemesis: "Zombies" }, { character: "Glory", nemesis: "Buffy" }, { character: "Bob", nemesis: "foo" }])</code> should return <code>[{"A_age": 27,"A_name": "Jonah", "B_character": "Jonah", "B_nemesis": "Whales"}, {"A_age": 27,"A_name": "Jonah", "B_character": "Jonah", "B_nemesis": "Spiders"}, {"A_age": 18,"A_name": "Alan", "B_character": "Alan", "B_nemesis": "Ghosts"}, {"A_age": 18,"A_name": "Alan", "B_character": "Alan", "B_nemesis": "Zombies"}, {"A_age": 28,"A_name": "Glory", "B_character": "Glory", "B_nemesis": "Buffy"}, {"A_age": 28,"A_name": "Alan", "B_character": "Alan", "B_nemesis": "Ghosts"}, {"A_age": 28,"A_name": "Alan", "B_character": "Alan", "B_nemesis": "Zombies"}]</code>'');'
+    testString: 'assert.deepEqual(hashJoin(hash1, hash2), res);'
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-join.english.md]:
```suggestion
    testString: assert.deepEqual(hashJoin(hash1, hash2), res);
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md [hunk 2]
```diff
@@ -26,17 +26,17 @@ tests:
   - text: <code>arrToObj</code> is a function.
     testString: assert(typeof arrToObj === 'function', '<code>arrToObj</code> is a function.');
   - text: '<code>arrToObj([1, 2, 3, 4, 5], ["a", "b", "c", "d", "e"])</code> should return <code>{ 1: "a", 2: "b", 3: "c", 4: "d", 5: "e" }</code>'
-    testString: 'assert.deepEqual(arrToObj(...testCases[0]), res[0], ''<code>arrToObj([1, 2, 3, 4, 5], ["a", "b", "c", "d", "e"])</code> should return <code>{ 1: "a", 2: "b", 3: "c", 4: "d", 5: "e" }</code>'');'
+    testString: 'assert.deepEqual(arrToObj(...testCases[0]), res[0]);'
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md]:
```suggestion
    testString: assert.deepEqual(arrToObj(...testCases[0]), res[0]);
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md [hunk 3]
```diff
@@ -26,17 +26,17 @@ tests:
...
   - text: '<code>arrToObj([1, 2, 3, 4, 5], ["a", "b", "c", "d"])</code> should return <code>{ 1: "a", 2: "b", 3: "c", 4: "d", 5: undefined }</code>'
-    testString: 'assert.deepEqual(arrToObj(...testCases[1]), res[1], ''<code>arrToObj([1, 2, 3, 4, 5], ["a", "b", "c", "d"])</code> should return <code>{ 1: "a", 2: "b", 3: "c", 4: "d", 5: undefined }</code>'');'
+    testString: 'assert.deepEqual(arrToObj(...testCases[1]), res[1]);'
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md]:
```suggestion
    testString: assert.deepEqual(arrToObj(...testCases[1]), res[1]);
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md [hunk 4]
```diff
@@ -26,17 +26,17 @@ tests:
...
   - text: '<code>arrToObj([1, 2, 3], ["a", "b", "c", "d", "e"])</code> should return <code>{ 1: "a", 2: "b", 3: "c" }</code>'
-    testString: 'assert.deepEqual(arrToObj(...testCases[2]), res[2], ''<code>arrToObj([1, 2, 3], ["a", "b", "c", "d", "e"])</code> should return <code>{ 1: "a", 2: "b", 3: "c" }</code>'');'
+    testString: 'assert.deepEqual(arrToObj(...testCases[2]), res[2]);'
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md]:
```suggestion
    testString: assert.deepEqual(arrToObj(...testCases[2]), res[2]);
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md [hunk 5]
```diff
@@ -26,17 +26,17 @@ tests:
...
   - text: '<code>arrToObj(["a", "b", "c", "d", "e"], [1, 2, 3, 4, 5])</code> should return <code>{ "a": 1, "b": 2, "c": 3 , "d": 4, "e": 5 }</code>'
-    testString: 'assert.deepEqual(arrToObj(...testCases[3]), res[3], ''<code>arrToObj(["a", "b", "c", "d", "e"], [1, 2, 3, 4, 5])</code> should return <code>{ "a": 1, "b": 2, "c": 3 , "d": 4, "e": 5 }</code>'');'
+    testString: 'assert.deepEqual(arrToObj(...testCases[3]), res[3]);'
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md]:
```suggestion
    testString: assert.deepEqual(arrToObj(...testCases[3]), res[3]);
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md [hunk 6]
```diff
@@ -26,17 +26,17 @@ tests:
...
   - text: '<code>arrToObj(["a", "b", "c", "d", "e"], [1, 2, 3, 4])</code> should return <code>{ "a": 1, "b": 2, "c": 3 , "d": 4, "e": undefined }</code>'
-    testString: 'assert.deepEqual(arrToObj(...testCases[4]), res[4], ''<code>arrToObj(["a", "b", "c", "d", "e"], [1, 2, 3, 4])</code> should return <code>{ "a": 1, "b": 2, "c": 3 , "d": 4, "e": undefined }</code>'');'
+    testString: 'assert.deepEqual(arrToObj(...testCases[4]), res[4]);'
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md]:
```suggestion
    testString: assert.deepEqual(arrToObj(...testCases[4]), res[4]);
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md [hunk 7]
```diff
@@ -26,17 +26,17 @@ tests:
...
   - text: '<code>arrToObj(["a", "b", "c"], [1, 2, 3, 4, 5])</code> should return <code>{ "a": 1, "b": 2, "c": 3  }</code>'
-    testString: 'assert.deepEqual(arrToObj(...testCases[5]), res[5], ''<code>arrToObj(["a", "b", "c"], [1, 2, 3, 4, 5])</code> should return <code>{ "a": 1, "b": 2, "c": 3  }</code>'');'
+    testString: 'assert.deepEqual(arrToObj(...testCases[5]), res[5]);'
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-from-two-arrays.english.md]:
```suggestion
    testString: assert.deepEqual(arrToObj(...testCases[5]), res[5]);
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/harshad-or-niven-series.english.md [hunk 8]
```diff
@@ -25,7 +25,7 @@ tests:
   - text: <code>isHarshadOrNiven</code> is a function.
     testString: assert(typeof isHarshadOrNiven === 'function', '<code>isHarshadOrNiven</code> is a function.');
   - text: '<code>isHarshadOrNiven()</code> should return <code>{"firstTwenty": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 18, 20, 21, 24, 27, 30, 36, 40, 42],"firstOver1000": 1002}</code>'
-    testString: 'assert.deepEqual(isHarshadOrNiven(), res, ''<code>isHarshadOrNiven()</code> should return <code>{"firstTwenty": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 18, 20, 21, 24, 27, 30, 36, 40, 42],"firstOver1000": 1002}</code>'');'
+    testString: 'assert.deepEqual(isHarshadOrNiven(), res);'
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/harshad-or-niven-series.english.md]:
```suggestion
    testString: assert.deepEqual(isHarshadOrNiven(), res);
```

**Review by thecodingaviator — CHANGES_REQUESTED** [review]
Please remove all these `'`, I'm not sure if they make a difference, but they taint the consistency

**Comment by RandellDawson** [comment]
@thecodingaviator Your request to remove the single quotes is beyond the scope of this PR, but I will do it on this one. If you want more of these done like this, then I think a new PR would be in order.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-join.english.md [hunk 9]
```diff
@@ -175,7 +175,7 @@ tests:
   - text: <code>hashJoin</code> is a function.
     testString: assert(typeof hashJoin === 'function', '<code>hashJoin</code> is a function.');
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-join.english.md]:
```suggestion
    testString: assert(typeof hashJoin === 'function');
```

Comment (thecodingaviator) [reply curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-join.english.md]: I must've missed this one mistakenly yesterday

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md [hunk 10]
```diff
@@ -25,11 +25,11 @@ tests:
...
+    testString: "assert((function() { var test = false; if (typeof BinarySearchTree !== 'undefined') { test = new BinarySearchTree() } else { return false; }; test.add(5); test.add(3); test.add(7); test.add(6); test.add(10); test.add(12); test.remove(3); test.remove(12); test.remove(10); return (typeof test.remove == 'function') ? (test.inorder().join('') == '567') : false})());"
   - text: The <code>remove</code> method removes nodes with one child.
     testString: assert((function() { var test = false; if (typeof BinarySearchTree !== 'undefined') { test = new BinarySearchTree() } else { return false; }; if (typeof test.remove !== 'function') { return false; }; test.add(-1); test.add(3); test.add(7); test.add(16); test.remove(16); test.remove(7); test.remove(3); return (test.inorder().join('') == '-1'); })(), 'The <code>remove</code> method removes nodes with one child.');
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md]:
```suggestion
    testString: assert((function() { var test = false; if (typeof BinarySearchTree !== 'undefined') { test = new BinarySearchTree() } else { return false; }; if (typeof test.remove !== 'function') { return false; }; test.add(-1); test.add(3); test.add(7); test.add(16); test.remove(16); test.remove(7); test.remove(3); return (test.inorder().join('') == '-1'); })());
```

Comment (thecodingaviator) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md]: I can't add reviews on lines 24, 26, 36, 38, and 40 but it's the same there

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md [hunk 11]
```diff
@@ -32,7 +32,7 @@ tests:
   - text: The Map data structure exists.
     testString: assert((function() { var test = false; if (typeof Map !== 'undefined') { test = new Map() }; return (typeof test == 'object')})(), 'The Map data structure exists.');
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md]:
```suggestion
    testString: assert((function() { var test = false; if (typeof Map !== 'undefined') { test = new Map() }; return (typeof test == 'object')})());
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md [hunk 12]
```diff
@@ -32,7 +32,7 @@ tests:
...
-    testString: "assert((function() { var test = false; if (typeof Map !== 'undefined') { test = new Map() }; return (typeof test.add == 'function' && typeof test.remove == 'function' && typeof test.get == 'function' && typeof test.has == 'function' && typeof test.values == 'function' && typeof test.clear == 'function' && typeof test.size == 'function')})(), 'The Map object has the following methods: add, remove, get, has, values, clear, and size.');"
+    testString: "assert((function() { var test = false; if (typeof Map !== 'undefined') { test = new Map() }; return (typeof test.add == 'function' && typeof test.remove == 'function' && typeof test.get == 'function' && typeof test.has == 'function' && typeof test.values == 'function' && typeof test.clear == 'function' && typeof test.size == 'function')})());"
   - text: The add method adds items to the map.
     testString: assert((function() { var test = false; if (typeof Map !== 'undefined') { test = new Map() }; test.add(5,6); test.add(2,3); test.add(2,5); return (test.size() == 2)})(), 'The add method adds items to the map.');
```
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md]:
```suggestion
    testString: assert((function() { var test = false; if (typeof Map !== 'undefined') { test = new Map() }; test.add(5,6); test.add(2,3); test.add(2,5); return (test.size() == 2)})());
```

Comment (thecodingaviator) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md]: Lines 41, 43, 45 need to be cleared too

**Review by thecodingaviator — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md (hunk 10 above)
Comment (RandellDawson) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/delete-a-node-with-two-children-in-a-binary-search-tree.english.md]: This PR only removed assert message arguments which were not removed in the original two Coding Interview Prep PRs (#36410 and #36412).  If you look at those PRs, you will find these were removed on one of them.

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md (hunk 11 above)
Comment (RandellDawson) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md]: This PR only removed assert message arguments which were not removed in the original two Coding Interview Prep PRs (#36410 and #36412).  If you look at those PRs, you will find this is removed on one of them.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-join.english.md (hunk 9 above)
Comment (RandellDawson) [reply curriculum/challenges/english/08-coding-interview-prep/rosetta-code/hash-join.english.md]: This PR only removed assert message arguments which were not removed in the original two Coding Interview Prep PRs (#36410 and #36412).  If you look at those PRs, you will find this is removed on one of them.

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md (hunk 12 above)
Comment (RandellDawson) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-map-data-structure.english.md]: This PR only removed assert message arguments which were not removed in the original two Coding Interview Prep PRs (#36410 and #36412). If you look at those PRs, you will find these are 
 removed on one of them.

**Review by thecodingaviator — APPROVED** [review]
LGTM

**Comment by thecodingaviator** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**